src/renderer/apis/settings.ts

```diff
@@ -1,64 +1,169 @@
+  /**
+   * Returns a copy of the settings data stored in this manager.
+   * @returns Current values of all settings in this manager's namespace.
+   */
+  public all(): T {
+    return { ...this.#settings } as T;
+  }
 
-    try {
-      res = await handler(settings);
-    } catch (e) {
-      await window.RepluggedNative.settings.endTransaction(this.namespace, null);
-      throw e;
+  #queueUpdate<K extends Extract<keyof T, string>>(key: K, update: SettingsUpdate<T>): void {
+    if (typeof this.#saveTimeout !== "undefined") {
+      clearTimeout(this.#saveTimeout);
     }
-
-    await window.RepluggedNative.settings.endTransaction(this.namespace, settings);
-    return res;
+    this.#queuedUpdates.set(key, update);
+    this.#saveTimeout = setTimeout(() => {
+      this.#queuedUpdates.forEach((u, k) => {
+        if (u.type === "delete") {
+          void window.RepluggedNative.settings.delete(this.namespace, k);
+        } else {
+          void window.RepluggedNative.settings.set(this.namespace, k, u.value);
+        }
+      });
+      this.#queuedUpdates.clear();
+      this.#saveTimeout = void 0;
+    }); // Add a delay of 1 or 2 seconds?
   }
 }
 
-export class SettingsAPI extends API {
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  public settings: Record<string, NamespacedSettings<any>> = {};
-
-  public constructor() {
-    super("dev.replugged.apis.Settings", "Settings");
-  }
+// I hope there's some way to force TypeScript to accept this, but for now unknown will do
+const managers = new Map<string, unknown>();
 
-  public get<T extends Settings>(namespace: string): NamespacedSettings<T> {
-    if (!this.settings[namespace]) {
-      this.settings[namespace] = new NamespacedSettings<T>(namespace);
-    }
-    return this.settings[namespace];
+/**
+ * Creates, initializes, and returns a {@link SettingsManager} for the given settings namespace. If a manager for the namespace already exists,
+ * then that instance will be returned. Use this function rather than creating instances of `SettingsManager` directly.
+ *
+ * Settings are stored synchronously in the window, and updates are dispatched asynchronously to the file system.
+ * See {@link SettingsManager} for more information on how this works.
+ *
+ * Here's an example of how to use this in a plugin:
+ *
+ * ```js
+ * import { settings } from 'replugged';
+ *
+ * const cfg = await settings.init('dev.replugged.Example');
+ *
+ * export function start() {
+ *   cfg.set('hello', 'world');
+ *   console.log(cfg.get('hello')); // world
+ * }
+ * ```
+ * @typeParam T Type definition for the settings to manage in the namespace.
+ * This will be an object with strings as keys, and JSON-serializable values.
+ * @param namespace Namespace to manage. A namespace is an ID (for example, the ID of a plugin) that uniquely identifies it.
+ * All settings are grouped into namespaces.
+ * Settings for a namespace are stored in `replugged/settings/NAMESPACE.json`.
```

Comment:
Clarify where it is stored
```suggestion
 * Settings for a namespace are stored in `settings/NAMESPACE.json` within the [Replugged data folder](https://docs.replugged.dev/#installing-plugins-and-themes).
```